presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1065,10 +1065,8 @@ protected Object visitLikePredicate(LikePredicate node, Object context)
                 return new ComparisonExpression(ComparisonExpressionType.EQUAL, valueExpression, patternExpression);
             }
 
-            Expression optimizedEscape = null;
-            if (node.getEscape() != null) {
-                optimizedEscape = toExpression(escape, type(node.getEscape()));
-            }
+            Optional<Expression> optimizedEscape = node.getEscape()
+                    .map(e -> toExpression(e, type(node.getEscape().get())));
```

Comment:
replace `type(node.getEscape().get())` with `type(e)` 

rename `e` to `escape`

Reply:
Unfortunately, `escape` conflicts with another variable in the method.

Reply:
In any case, the logic was wrong. After my change, it lost the ability to optimize the escape pattern.